barcode/ean.py

```diff
@@ -54,14 +56,9 @@ def __init__(self, ean, writer=None, no_checksum=False):
                     len(ean),
                 )
             )
-        self.ean = ean
-        # If no checksum
+
         if no_checksum:
-            # Add a thirteen char if given in parameter,
-            # otherwise pad with zero
-            self.ean = "{}{}".format(
-                ean, ean[self.digits] if len(ean) > self.digits else 0
-            )
+            self.ean = ean
```

Comment:
The original `ean` value is lost due to the `ean = ean[: self.digits]` statement above.

This won't work.